### PR TITLE
266 remove backward compatibility with khiops 9

### PR DESCRIPTION
**Pull request by folmos-at-orange**
```
commit 8d22bb6b320453220a01557b60f98a52e2478bd6 (HEAD -> 266-remove-backward-compatibility-with-khiops-9, origin/266-remove-backward-compatibility-with-khiops-9)
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Fri Nov 22 15:22:08 2024 +0100

    Update CHANGELOG.md

commit 1dc8be55969bdb72414c67296df77d7c914fa051
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Fri Nov 22 15:18:49 2024 +0100

    Remove `pykhiops`-> `khiops` compatibility code

commit eb5915df71dffe664746a37d5271e361fb3ec784
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Thu Oct 24 14:17:22 2024 +0200

    Remove Khiops 9 scenario exit statement

commit 593f2665c07d2d13fcc9bfeeff9b85d24aa17930
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Fri Nov 22 15:24:27 2024 +0100

    Bump min_version of tasks from 9 to 10

commit 3804c96d99c107a23c776aed2d2b73949d2f4238
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Thu Oct 24 14:16:41 2024 +0200

    Remove Khiops 9 scenario templates

commit 4d95f652ae5ef631599727f8e39bf137f3247692
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Thu Oct 24 14:16:05 2024 +0200

    Remove pykhiops 9 removal/rename warnings

commit 32adf243fed69ebc4ffa368faf656e1e359c1d91
Author: Felipe Olmos <92923444+folmos-at-orange@users.noreply.github.com>
Date:   Thu Oct 24 14:14:26 2024 +0200

    Remove old get_tool_info functions

    Remove Khiops 9 and Khiops < 10.1 tool info functions.
```

### CODE REVIEW

File: khiops/core/internals/runner.py [hunk 1]
```diff
@@ -1100,14 +1063,13 @@ def _initialize_khiops_version(self):
                 if line.startswith("Khiops"):
                     khiops_version_str = line.rstrip().split(" ")[1]
                     break
-        # If -v fails it means it is Khiops 9 or lower so we try the old way
+        # If -v fails the environment is KO: raise an error
```
Comment (popescu-v) [inline khiops/core/internals/runner.py:1066]: IMHO, it could fail for other reasons, e.g. the executable at `runner.khiops_path` fails to run.

File: khiops/core/internals/runner.py [hunk 2]
```diff
@@ -1100,14 +1063,13 @@ def _initialize_khiops_version(self):
...
         else:
-            khiops_version_str, _, _, _ = _get_tool_info_khiops9(self, "khiops")
-            warnings.warn(
-                "Khiops version is earlier than 10.0; the Khiops Python library will "
-                f"run in legacy mode. Khiops path: {self._tool_path('khiops')}",
-                stacklevel=3,
-            )
+            error_msg = f"Could not execute 'khiops -v': return code {return_code}."
```
Comment (popescu-v) [inline khiops/core/internals/runner.py:1068]: IMHO, we mustn't mention `khiops -v` as this means nothing to the user and, moreover, this is not true ("khiops" here is just a lookup key for `runner._tool_path`): we execute whatever is at `runner.khiops_path`, which is rather MODL / MODL_openmpi / MODL_mpich, but this is also transparent to the user.

So, I would rephrase to something like:

```python
error_msg = f"Could not get Khiops version: Khiops binary return code {return_code}"
```

File: khiops/core/internals/runner.py [hunk 3]
```diff
@@ -1100,14 +1063,13 @@ def _initialize_khiops_version(self):
...
+            error_msg += f"\nstdout: {stdout}" if stdout else ""
+            error_msg += f"\nstderr: {stderr}" if stderr else ""
+            raise KhiopsEnvironmentError(error_msg)
```
Comment (popescu-v) [inline khiops/core/internals/runner.py:1071]: I would use `KhiopsRuntimeError` instead (see related comment.s above).

**Review by popescu-v — CHANGES_REQUESTED** [review]
Some changes (see comments), plus:

- I would also consider removing the pykhiops / PyKhiops deprecations, as they do not concern recent v10 versions anymore (as any version >= 10.2.3.0 is not backward compatible with previous versions supporting the pykhiops syntax anyway)
- I would also consider bumping to "10.0.0" the remaining "9.0.0" in the few `KhiopsTask`s concerned, namely:
  - `extract_clusters`
  - `build_multi_table_dictionary`
  - `prepare_coclustering_deployment`
  - `export_dictionary_as_json`
  
  as keeping this "9.0.0" doesn't bring any useful information now that we gave up supporting "n-1" Khiops versions.
  And I would consider removing this task versioning alotogether, but perhaps in a different PR.

File: khiops/core/internals/runner.py (hunk 2 above)
Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:1068]: Ok I'll rephrase but I'll add all the necessary info to diagnose.

Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:1068]: Done

File: khiops/core/internals/runner.py (hunk 1 above)
Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:1066]: Ok & Done

File: khiops/core/internals/runner.py (hunk 3 above)
Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:1071]: Ok & Done

File: khiops/core/internals/runner.py (hunk 2 above)
Comment (popescu-v) [reply khiops/core/internals/runner.py:1068]: The message is visible to the users of the library, via the public accessor `khiops_version` of the `KhiopsLocalRunner`. Therefore, IMHO it should be legible to the users.

Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:1068]: I made the first line legible for the users. The rest are diagnostics for us.

**Review by popescu-v — APPROVED** [review]
